routers/web/repo/editor.go

```diff
@@ -420,7 +420,7 @@ func DiffPreviewPost(ctx *context.Context) {
 	}
 
 	if diff.NumFiles == 0 {
-		ctx.PlainText(http.StatusOK, ctx.Locale.TrString("repo.editor.no_changes_to_show"))
+		ctx.PlainText(http.StatusOK, `<div class="tw-p-6">` + ctx.Locale.TrString("repo.editor.no_changes_to_show") + `</div>`)
```

Comment:
This does return `content-type: text/plain;charset=utf-8` but still works, fwiw. `ctx.HTML` is unsuitable as I can not pass a string to it.

Reply:
https://github.com/go-gitea/gitea/pull/30707/commits/7e401494155bc3a5fd78d66158f1ee62de13c8ed adds `ctx.HTMLString`, if you have better ideas, please shoot :)

Reply:
@wxiaoguang any suggestions on how to avoid adding this `HTMLString` function?

Reply:
No need to do so. Just use `{{if eq NumFiles 0}} ... {{end}}` in the template, and do not return early.

Reply:
Why resolve?